### PR TITLE
Use custom frequency for telemetry and device state

### PR DESCRIPTION
**Pull request by dluc**
# Type of change? 

- [ ] Bug fix
- [x] New feature
- [ ] Enhancement
- [ ] Breaking change (breaks backward compatibility)

# Description, Context, Motivation 

Apply the user customizations to the device models, e.g. custom telemetry frequency and custom device state simulation frequency.
Fix a regression affecting device models with multiple telemetry messages.

**Checklist:**

- [x] All tests passed
- [x] The code follows the code style and conventions of this project
- [ ] The change requires a change to the documentation
- [ ] I have updated the documentation accordingly

---
This change is [Reviewable](https://reviewable.io/reviews/azure/device-simulation-dotnet/141)

### CODE REVIEW

File: SimulationAgent/DeviceTelemetry/DeviceTelemetryActor.cs [hunk 1]
```diff
@@ -163,28 +163,31 @@ public void HandleEvent(ActorEvents e)
...
 
             var now = DateTimeOffset.UtcNow.ToUnixTimeMilliseconds();
-            if (now < this.whenToRun) return;
+            if (now < this.whenToRun) return null;
 
             switch (this.status)
             {
                 case ActorStatus.ReadyToStart:
-                    if (!this.deviceConnectionActor.Connected) return;
+                    if (!this.deviceConnectionActor.Connected) return null;
```
Comment (timlaverty) [inline SimulationAgent/DeviceTelemetry/DeviceTelemetryActor.cs]: return "device not connected"

File: Services/DeviceModels.cs [hunk 2]
```diff
@@ -78,6 +79,89 @@ public DeviceModel Get(string id)
             throw new ResourceNotFoundException();
         }
 
+        public DeviceModel OverrideDeviceModel(DeviceModel source, Models.Simulation.DeviceModelOverride overrideInfo)
```
Comment (timlaverty) [inline Services/DeviceModels.cs]: do we have a github issue tracking revisiting the design of the override classes?  The more I think of it the more I think they're going to get unwieldy & hard to maintain in the future.

File: SimulationAgent/DeviceTelemetry/DeviceTelemetryActor.cs (hunk 1 above)
Comment (dluc) [reply SimulationAgent/DeviceTelemetry/DeviceTelemetryActor.cs]: 👍

File: Services/DeviceModels.cs (hunk 2 above)
Comment (dluc) [reply Services/DeviceModels.cs]: https://github.com/Azure/device-simulation-dotnet/issues/144

**Pull request merged**